### ISSUE TITLE
CrossTalk Operator doesn't run on >3 processors 

### ISSUE THREAD

**Issue by giuspugl** [hpc4cmb/toast#590]
I have been further testing the [CrossTalk operator ](https://github.com/hpc4cmb/toast/blob/ab4e1794abb7069e30f01b46c07a782c39842d87/src/toast/ops/sim_crosstalk.py#L126and) its related [tests](https://github.com/hpc4cmb/toast/blob/toast3/src/toast/tests/ops_sim_crosstalk.py) and found out that fetching the  latest changes in toast  ( the crosstalk PR is ~ 1yr old https://github.com/hpc4cmb/toast/pull/421 )  something has   broken the operator and that is visible when running the test on ,say, 4 processing elements or more.  I get the following error:

```
  KeyError: <MPIShared
  replicated on 1 nodes, each with 2 processes (2 total)
  shape = (6000,), dtype = float64
  [ 1677110400.0 1677110400.1 ... 1677110999.8000002 1677110999.9 ]
```

This is happening exactly in the line we perform the redistribution i.e. https://github.com/hpc4cmb/toast/blob/ab4e1794abb7069e30f01b46c07a782c39842d87/src/toast/ops/sim_crosstalk.py#L283 
 any  clue   ?

**Comment by tskisner** [comment]
I believe this was fixed in #593 .  Running with latest toast3 branch (using a conda environment), I first created a conda env for installing toast:
```
# Verify that conda is available
$ which conda
/home/kisner/software/conda/condabin/conda
```
Setup the conda env:
```
cd toast
./platforms/conda_dev_setup.sh toast_test
```
Activate that env:
```
conda activate toast_test
```
Install mpi4py.  On most laptops / workstations this comes with a bundled openmpi that just works:
```
(toast_test) $  conda install mpi4py
```
Install toast from source as a pip package, using the conda compilers:
```
(toast_test) $  ./platforms/conda.sh
```
Now run the tests in question:
```
(toast_test) $  python -c 'import toast; print(toast.__version__)'
3.0.0a8

(toast_test) $  OMP_NUM_THREADS=2 mpirun -np 4 python -c 'import toast.tests; toast.tests.run("ops_sim_crosstalk")'

test_xtalk (toast.tests.ops_sim_crosstalk.SimCrossTalkTest) ... TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
[1]ok [0]ok 
test_xtalk_big (toast.tests.ops_sim_crosstalk.SimCrossTalkTest) ... [2]ok [3]ok TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
[1]ok [0]ok [3]ok [2]ok 
test_xtalk_errors (toast.tests.ops_sim_crosstalk.SimCrossTalkTest) ... TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
[2]ok [1]ok [0]ok 
[3]ok test_xtalk_matrices (toast.tests.ops_sim_crosstalk.SimCrossTalkTest) ... TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
TOAST WARNING: Mission start time '2023-02-23 00:00:00' is not timezone-aware.  Assuming UTC.
[1]ok [0]ok [2]ok [3]ok 

----------------------------------------------------------------------
Ran 4 tests in 0.268s

[0] OK
[1] OK
[2] OK
[3] OK
```

**Comment by giuspugl** [comment]
Thanks a lot for the headsup @tskisner ! Now test  perfectly runs ! 
 however it 's not clear how you fixed it in PR#593  ?
